geoh5py/objects/geo_image.py

```diff
@@ -238,3 +280,181 @@ def vertices(self, xyz: np.ndarray | list):
         )
         self._vertices = xyz
         self.workspace.update_attribute(self, "vertices")
+
+    @property
+    def tag(self) -> dict | None:
+        """
+        Return the tag information of the input image (if .tif).
+        :return: a dictionary containing the PIL.Image.tag information.
+        """
+        if self._tag:
+            return self._tag.copy()
+        return None
+
+    @tag.setter
+    def tag(self, image: Image.Image | dict | None):
+        if isinstance(image, (Image.Image, TiffImageFile)):
+            self._tag = dict(image.tag)
+        elif isinstance(image, dict):
+            self._tag = image
+        elif image is None:
+            self._tag = None
+        else:
+            raise ValueError("Input 'tag' must be a PIL.Image")
+
+    def georeferencing_from_tiff(self):
+        """
+        Get the geogrpahic information from the PIL Image to georeference it.
+        Run the georefence() method of the object.
+        :param image: a .tif image open with PIL.Image.
+        """
+        try:
+            if self._tag:
+                # get geographic information
+                u_origin = self._tag[33922][3]
+                v_origin = self._tag[33922][4]
+                u_cell_size = self._tag[33550][0]
+                v_cell_size = self._tag[33550][1]
+                u_count = self._tag[256][0]
+                v_count = self._tag[257][0]
+                u_oposite = u_origin + u_cell_size * u_count
+                v_oposite = v_origin - v_cell_size * v_count
+
+                # prepare georeferencing
+                reference = np.array(
+                    [[0.0, v_count], [u_count, v_count], [u_count, 0.0]]
+                )
+
+                locations = np.array(
+                    [
+                        [u_origin, v_origin, 0.0],
+                        [u_oposite, v_origin, 0.0],
+                        [u_oposite, v_oposite, 0.0],
+                    ]
+                )
+
+                # georeference the raster
+                self.georeference(reference, locations)
+
+        except (AttributeError, KeyError):
```

Comment:
Can remove the `AttributeError` from this `try`, and just be more specific with logic for `if self.tag is None`?